.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,53 @@
+name: End-to-end tests
+
+on:
+  push:
+    branches: [ development ]
+    paths-ignore: [ '*.md']
+  pull_request:
+    branches: [ '**' ]
+  merge_group:
+
+jobs:
+  test:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres
+
+        env:
+          POSTGRES_PASSWORD: dbpwd
+          POSTGRES_USER: dbuser
+          POSTGRES_DB: pearl
+
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'adopt'
+    - name: Setup Gradle
+      uses: gradle/gradle-build-action@v2
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
+    - name: Install dependencies
+      run: npm ci
+    - name: Build applications
+      run: |
+        ./gradlew :api-admin:build -x test
+        ./gradlew :api-admin:copyWebApp
+        ./gradlew :api-participant:build -x test
+        ./gradlew :api-participant:copyWebApp
```

Comment:
I imagine we're building the applications in other actions. Does this action need to be isolated or can it use artifacts produced by another action? Or is there something special about the artifacts needed for the e2e tests?

Reply:
This is currently the only place where we build the UI into the Java app. We could possibly combine all CI into one workflow and go: build JS => build Java => run e2e tests.